### DISCUSSION TITLE
MySQL Consultation

### DISCUSSION THREAD

**Discussion by shitizz24** [wgu-opensource/osmt#193] [Helps to know] [answered]
We want to onboard OSMT into our organisation's ecosystem and have the following MySQL offerings available in our organisation, DBEvolver, GCP CloudSQL MySQL, and Cloud RDBMS.
wanted to know what all of these offerings would be compatible with OSMT
Thanks,
Shitiz

**Comment by JohnKallies** [comment, marked as answer]
Hi @shitizz24.

I'm happy to see you bringing OSMT in to the fold.

I'm not a MySQL expert in the enterprise space, and I don't have an experience with the products you mentioned. Most of my experience with the platform comes from much looser projects. That said, can I reframe your question? I hear you asking about how to use enterprise MySQL resources and services in support of your OSMT deployment. Please redirect me if I don't have your question right?

There's nothing especially tricky about how OSMT uses the MySQL database. The Spring application uses Flyway to apply the DDL scripts in the `api/src/main/resources/db/migration` directory, from oldest to newest. Any number of tools can get the `osmt_db` schema from an OSMT instance (including a Quickstart) and create a DDL script, or reapply it to a different server, or even port it to a different platform.

I think your question probably defines the bounds for some spike work for your organization. For what it's worth, if I were deploying OSMT in a cloud context, I would favor a managed database service.

I hope that helps...
John

**Reply by shitizz24** [reply]
Okay,Thanks @JohnKallies.Looks like we might have to do some analysis on our own to choose entreprise MySQL resources for OSMT deployment.
One more thing, i came to know that currently, my organisation walmart isn't onboarding any new redis resources unless there is an exception invloved,in this case would OSMT be compatible with Meghacache?

**Reply by JohnKallies** [reply]
Hi @shitizz24. I was out for a couple days, thanks for your patience. I'm not picking at your typing, but I don't understand what product you are pointing to above. Do you mean memcached?

**Reply by JohnKallies** [reply]
Assuming you meant memcache above, I have no experience using memcache, and no insights regarding how compatible the API might be, or what abstractions might exist.

If you search OSMT's API module, you'll find a couple key places where Redis is tightly coupled. 
- TaskMessageService.kt (which is tightly-coupled to a Spring RedisRestTemplate, and to RQueueMessageSender, from https://github.com/sonus21/rqueue)
- RedisConfig.kt

One could implement and inject the needed abstractions for some alternative in-memory database/cache provider, but in the containerized/microserviced point of view, the existing implementation seems pretty opinionated for Redis. It doesn't need a shared enterprise Redis, but rather just one for it's own app stack. I personally think that reimplementing for an abstraction would not bring the same value as just deploying a Redis container in whatever context the OSMT app and service run in.

This may help share some of the intent behind the current implementation...
- https://spring.io/projects/spring-data-redis

I hope this helps!
John

**Reply by shitizz24** [reply]
You got it right,John. Came to know Meghacache is just an enterprise name for Memcache. 
Looks like we got to do some spike work on our own to spin up OSMT with available enterprise products.
Thanks for the documentation references ,it really helps.
